hyperspy/signal.py

```diff
@@ -3485,35 +3482,58 @@ def unfold(self):
         """Modifies the shape of the data by unfolding the signal and
         navigation dimensions separately
 
+        Returns
+        -------
+        needed_unfolding : bool
+
+
         """
-        self.unfold_navigation_space()
-        self.unfold_signal_space()
+        nav_needed_unfolding = self.unfold_navigation_space()
+        sig_needed_unfolding = self.unfold_signal_space()
+        needed_unfolding = nav_needed_unfolding or sig_needed_unfolding
+        return needed_unfolding
 
     def unfold_navigation_space(self):
         """Modify the shape of the data to obtain a navigation space of
         dimension 1
+
+        Returns
+        -------
+        needed_unfolding : bool
+
         """
 
         if self.axes_manager.navigation_dimension < 2:
-            return False
-        steady_axes = [
-            axis.index_in_array for axis in
-            self.axes_manager.signal_axes]
-        unfolded_axis = (
-            self.axes_manager.navigation_axes[0].index_in_array)
-        self._unfold(steady_axes, unfolded_axis)
+            needed_unfolding = False
+        else:
+            needed_unfolding = True
+            steady_axes = [
+                axis.index_in_array for axis in
+                self.axes_manager.signal_axes]
+            unfolded_axis = (
+                self.axes_manager.navigation_axes[0].index_in_array)
+            self._unfold(steady_axes, unfolded_axis)
+            return needed_unfolding
 
     def unfold_signal_space(self):
         """Modify the shape of the data to obtain a signal space of
         dimension 1
+
+        Returns
+        -------
+        needed_unfolding : bool
+
         """
         if self.axes_manager.signal_dimension < 2:
-            return False
-        steady_axes = [
-            axis.index_in_array for axis in
-            self.axes_manager.navigation_axes]
-        unfolded_axis = self.axes_manager.signal_axes[0].index_in_array
-        self._unfold(steady_axes, unfolded_axis)
+            needed_unfolding = False
+        else:
+            needed_unfolding = True
+            steady_axes = [
+                axis.index_in_array for axis in
+                self.axes_manager.navigation_axes]
+            unfolded_axis = self.axes_manager.signal_axes[0].index_in_array
+            self._unfold(steady_axes, unfolded_axis)
+            return needed_unfolding
```

Comment:
Thanks, it should be fine now.
On Tue, 2015-07-21 at 05:13 -0700, vidartf wrote:
> In hyperspy/signal.py:
> >          """
> >          if self.axes_manager.signal_dimension < 2:
> > -            return False
> > -        steady_axes = [
> > -            axis.index_in_array for axis in
> > -            self.axes_manager.navigation_axes]
> > -        unfolded_axis =
> self.axes_manager.signal_axes[0].index_in_array
> > -        self._unfold(steady_axes, unfolded_axis)
> > +            needed_unfolding = False
> > +        else:
> > +            needed_unfolding = True
> > +            steady_axes = [
> > +                axis.index_in_array for axis in
> > +                self.axes_manager.navigation_axes]
> > +            unfolded_axis =
> self.axes_manager.signal_axes[0].index_in_array
> > +            self._unfold(steady_axes, unfolded_axis)
> > +            return needed_unfolding
> This one too.
> —
> Reply to this email directly or view it on GitHub.